schema_based_on_diagram.sql

```diff
@@ -0,0 +1,98 @@
+-- Table: public.treatments
+
+-- DROP TABLE IF EXISTS public.treatments;
+
+CREATE TABLE IF NOT EXISTS public.treatments
+(
+    id integer NOT NULL,
+    type character varying COLLATE pg_catalog."default",
+    name character varying COLLATE pg_catalog."default",
+    CONSTRAINT treatments_pkey PRIMARY KEY (id)
+)
+
+TABLESPACE pg_default;
+
+ALTER TABLE IF EXISTS public.treatments
+    OWNER to postgres;
+CREATE TABLE IF NOT EXISTS public.patients
+(
+    id integer NOT NULL,
+    name character varying COLLATE pg_catalog."default" NOT NULL,
+    date_of_birth date NOT NULL,
+    CONSTRAINT patients_pkey PRIMARY KEY (id)
+)
+
+TABLESPACE pg_default;
+
+ALTER TABLE IF EXISTS public.patients
+    OWNER to postgres;
+	
+CREATE TABLE IF NOT EXISTS public.medical_histories
+(
+    id integer NOT NULL,
+    admitted_at timestamp without time zone,
+    patient_id integer,
+    status character varying COLLATE pg_catalog."default",
+    CONSTRAINT medical_histories_pkey PRIMARY KEY (id),
+    CONSTRAINT id FOREIGN KEY (id)
+        REFERENCES public.treatments (id) MATCH SIMPLE
+        ON UPDATE NO ACTION
+        ON DELETE NO ACTION
+        NOT VALID,
+    CONSTRAINT patient_id FOREIGN KEY (patient_id)
+        REFERENCES public.patients (id) MATCH SIMPLE
+        ON UPDATE NO ACTION
+        ON DELETE NO ACTION
+        NOT VALID
+)
+
+TABLESPACE pg_default;
+
+ALTER TABLE IF EXISTS public.medical_histories
+    OWNER to postgres;
+	
+CREATE TABLE IF NOT EXISTS public.invoices
+(
+    id integer NOT NULL,
+    total_amount numeric,
+    generated_at timestamp without time zone,
+    payed_at timestamp without time zone,
+    medical_history_id integer,
+    CONSTRAINT invoices_pkey PRIMARY KEY (id),
+    CONSTRAINT medical_history_id FOREIGN KEY (medical_history_id)
+        REFERENCES public.medical_histories (id) MATCH SIMPLE
+        ON UPDATE NO ACTION
+        ON DELETE NO ACTION
+        NOT VALID
+)
+
+TABLESPACE pg_default;
+
+ALTER TABLE IF EXISTS public.invoices
+    OWNER to postgres;
+
+CREATE TABLE IF NOT EXISTS public.invoice_items
+(
+    id integer NOT NULL,
+    unit_price numeric,
+    quantity integer,
+    total_price numeric,
+    invoice_id integer,
+    treatment_id integer,
+    CONSTRAINT invoice_items_pkey PRIMARY KEY (id),
+    CONSTRAINT invoice_id FOREIGN KEY (invoice_id)
+        REFERENCES public.invoices (id) MATCH SIMPLE
+        ON UPDATE NO ACTION
+        ON DELETE NO ACTION
+        NOT VALID,
+    CONSTRAINT treatment_id FOREIGN KEY (treatment_id)
+        REFERENCES public.treatments (id) MATCH SIMPLE
+        ON UPDATE NO ACTION
+        ON DELETE NO ACTION
+        NOT VALID
+)
+
+TABLESPACE pg_default;
+
+ALTER TABLE IF EXISTS public.invoice_items
+    OWNER to postgres;
```

Comment:
- Kindly create a table for the relation many-to-many between medical_histories and treatments.

```sql
CREATE TABLE medical_histories_has_treatments (
    medical_history_id int refrences medical_histories(id),
    treatment_id int refrences treatments(id),
);
```

- [ ] Also create indexes for all FOREIGN KEYS have indexes created.
* `CREATE INDEX ON medical_histories (patient_id);`
* `CREATE INDEX ON invoices (medical_history_id);`
* `CREATE INDEX ON invoice_items (invoice_id);`
* `CREATE INDEX ON invoice_items (treatment_id);`
* `CREATE INDEX ON medical_histories_has_treatments (medical_history_id);`
* `CREATE INDEX ON medical_histories_has_treatments (treatment_id);`

Reply:
@ShahierNashaat Hi. Thanks for your review. There's a relationship between them. You mean another?

CREATE TABLE IF NOT EXISTS public.medical_histories
(
    id integer NOT NULL,
    admitted_at timestamp without time zone,
    patient_id integer,
    status character varying COLLATE pg_catalog."default",
    CONSTRAINT medical_histories_pkey PRIMARY KEY (id),
    CONSTRAINT id FOREIGN KEY (id)

Reply:
Yes another relation

Reply:
Hi. Sorry to disagree with you. This are the two diagrams (from the github project and from my table). Please compare.
![Medical_Database_Diagram pgerd](https://user-images.githubusercontent.com/103516637/195200281-e7c055d5-4784-4ae4-91b3-96f55be90e0f.png)
![clinic_diagram](https://user-images.githubusercontent.com/103516637/195200397-b36ff569-4a25-4589-a959-7577ecf81dcb.png)